pkg/nuget/csproj/parse.go

```diff
@@ -0,0 +1,52 @@
+package csproj
+
+import (
+	"encoding/xml"
+
+	"golang.org/x/xerrors"
+
+	dio "github.com/aquasecurity/go-dep-parser/pkg/io"
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+	"github.com/aquasecurity/go-dep-parser/pkg/utils"
+)
+
+type cfgPackageReference struct {
+	XMLName       xml.Name `xml:"PackageReference"`
+	Version       string   `xml:"Version,attr"`
+	Include       string   `xml:"Include,attr"`
+	PrivateAssets string   `xml:"PrivateAssets"`
+}
+
+type config struct {
+	XMLName  xml.Name              `xml:"Project"`
+	Packages []cfgPackageReference `xml:"ItemGroup>PackageReference"`
+}
+
+type Parser struct{}
+
+func NewParser() types.Parser {
+	return &Parser{}
+}
+
+func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
+	var cfgData config
+	if err := xml.NewDecoder(r).Decode(&cfgData); err != nil {
+		return nil, nil, xerrors.Errorf("failed to decode .csproj file: %w", err)
+	}
+
+	libs := make([]types.Library, 0)
+	for _, pkg := range cfgData.Packages {
+		if pkg.Include == "" || pkg.PrivateAssets != "" {
+			continue
+		}
+
+		lib := types.Library{
+			Name:    pkg.Include,
+			Version: pkg.Version,
```

Comment:
I [changed the code](https://github.com/aquasecurity/go-dep-parser/pull/121/commits/2e57fbafe8d43d821a3a94a3d7f8aeb2c3877f95) to trim `.*`. I don't think it is elegant to reduce the version like this, but I guess it's the best option that's possible. It results in desired behavior in trivy, because if 6.0.4.5 is vulnerable, you don't want to specify 6.0.4.* as dependency.

Reply:
> because if 6.0.4.5 is vulnerable, you don't want to specify 6.0.4.* as dependency.
Agree with you!